lib/TPP/TransformUtils.cpp

```diff
@@ -184,6 +185,26 @@ FailureOr<SmallVector<Range>> getLoopsToMaterialize(RewriterBase &rewriter,
   return loopRanges;
 }
 
+bool isBlockedConvolution(linalg::LinalgOp linalgOp) {
+  auto iteratorTypes = linalgOp.getIteratorTypesArray();
+  if (iteratorTypes.size() != 9)
+    return false;
+  bool match = linalg::isParallelIterator(iteratorTypes[0]) &&
```

Comment:
nit: you could probably replace this check with `std::all_of` for better readability

Reply:
I check for parallel and reduction. why std::all_of?

Reply:
Just personal opinion 2 lines vs 9 - if you think it doesn't matter, feel free to ignore this.